db/migrations/20180917141421_users_table.js

```diff
@@ -0,0 +1,22 @@
+
+exports.up = function(knex, Promise) {
+  return knex.schema.createTable('users_table', (tbl) => {
+    
+    tbl
+    .increments()
+
+    tbl
+    .string('username', 128)
+    .notNullable()
```

Comment:
I recommend using `.unique()` here to ensure there are no duplicate usernames. I think that may cause problems with login if there are. Otherwise, this is looking nice. 👍

Reply:
That's awesome, I tried that with Rachel this morning because she had it, but after trying it both ways, it didn't seem to change the ability to create a user from my tests. I ended up not using the code since I realized I couldn't fully explain if I needed it or not.

Reply:
As long as there's no duplicates, you shouldn't see any problems. However `.unique()` _enforces_ no duplicates, whereas without it, there's no problems until someone has the same username as someone else. Then your `login` route will end up returning two people and mess that up.

Reply:
Or are you saying you tried to add duplicate users and `.unique()` did nothing?